### PR TITLE
Only destroy test specific caches at setup and teardown of SCC tests

### PR DESCRIPTION
**Pull request by Mesbah-Alam**
Only destroy test specific caches at setup and teardown of tests
Signed-off-by: Mesbah_Alam@ca.ibm.com <Mesbah_Alam@ca.ibm.com>

### CODE REVIEW

**Comment by Mesbah-Alam** [comment]
Related: https://github.com/eclipse/openj9-systemtest/issues/62

**Comment by Mesbah-Alam** [comment]
SharedClasses_Workload : https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/825/console
SharedClassesAPI: https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/826/console
SharedClasses_Softmx_Increase: https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/827/console
SharedClasses_Softmx_UpDownz; https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/828/
SharedClasses_Softmx_JitAot: https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/830/console
SharedClasses.SCM01.SingleCL: https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/831/console

**Comment by Mesbah-Alam** [comment]
@pshipton  - This PR updates all the Shared Classes tests to destroy only the test related caches at setup and teardown stages. I've verified the changes work based on specific tests. Could you please review and merge it?

File: openj9.test.sharedClasses.jvmti/src/test.sharedClasses.jvmti/net/openj9/stf/SharedClassesAPI.java [hunk 1]
```diff
@@ -114,10 +115,14 @@ public void setUp(StfCoreExtension test, StfSharedClassesExtension sharedClasses
...
 		test.doMkdir("Create the config directory", configDirLocation);
 		
-		// To ensure we run from a clean state, attempt to destroy all persistent/non-persistent caches 
-		// from the default cache location which may have been left behind by a previous failed test.
-		sharedClasses.doDestroyAllPersistentCaches("Destroy Persistent Shared Classes Caches");
-		sharedClasses.doDestroyAllNonPersistentCaches("Destroy Non-Persistent Shared Classes Caches");
+		//We are running 5 Workloads namely WL1,..WL4. Each creates its own cache, so we should clean up each one of them
+		for (Tests apiTest : Tests.values()) {
+			for (int i = 1 ; i < 5 ; i++) {
+				String cacheName = apiTest.name() + "WL1" + i;
```
Comment (pshipton) [inline openj9.test.sharedClasses.jvmti/src/test.sharedClasses.jvmti/net/openj9/stf/SharedClassesAPI.java]: Ideally the number of iterations and the cache name should be shared with the test and not hard coded separately. That way if the tests changes, the cleanup continues to be correct.

**Comment by pshipton** [comment]
Started a test build https://ci.eclipse.org/openj9/view/Pipelines/job/Pipeline-Release-Build/20/

File: openj9.test.sharedClasses.jvmti/src/test.sharedClasses.jvmti/net/openj9/stf/SharedClassesAPI.java (hunk 1 above)
Comment (Mesbah-Alam) [reply openj9.test.sharedClasses.jvmti/src/test.sharedClasses.jvmti/net/openj9/stf/SharedClassesAPI.java]: Agree. I'll fix this in a separate PR.

**Comment by Mesbah-Alam** [comment]
Related : https://github.com/eclipse/openj9-systemtest/issues/73

**Pull request merged**